### PR TITLE
Task Library: Migrate Embedder task APIs

### PR DESCRIPTION
**Pull request by kinaryml**

### CODE REVIEW

File: tensorflow_lite_support/python/task/processor/proto/embedding_options_pb2.py [hunk 1]
```diff
@@ -13,6 +13,57 @@
...
 """Embedding options protobuf."""
 
+import dataclasses
+from typing import Any, Optional
+
+from tensorflow.tools.docs import doc_controls
 from tensorflow_lite_support.cc.task.processor.proto import embedding_options_pb2
 
-EmbeddingOptions = embedding_options_pb2.EmbeddingOptions
+_EmbeddingOptions = embedding_options_pb2.EmbeddingOptions
```
Comment (khanhlvg) [inline tensorflow_lite_support/python/task/processor/proto/embedding_options_pb2.py:22]: Rename to _ EmbeddingOptionsProto to improve readability.

File: tensorflow_lite_support/python/task/processor/proto/embedding_pb2.py [hunk 2]
```diff
@@ -13,8 +13,144 @@
...
+  Feature vectors are assumed to be one-dimensional and L2-normalized.
+
+  Attributes:
+    value_float: Raw output of the embedding layer. Only provided if `quantize`
+      is set to False in the `EmbeddingOptions`, which is the case by default.
+    value_string: Scalar-quantized embedding. Only provided if `quantize` is
+      set to true in `EmbeddingOptions`.
+  """
+
+  value_float: List[float] = None
```
Comment (khanhlvg) [inline tensorflow_lite_support/python/task/processor/proto/embedding_pb2.py:39]: Due to more flexibility in Python, I think we can use a single attribute `value: np.ndarray` here. The datatype of elements in the array can be either np.single (float) or np.byte. You can look at the data type to convert it to protobuf's value_float and value_string accordingly.

File: tensorflow_lite_support/python/task/processor/proto/embedding_pb2.py [hunk 3]
```diff
@@ -13,8 +13,144 @@
...
+
+  def __eq__(self, other: Any) -> bool:
+    """Checks if this object is equal to the given object.
+    Args:
+      other: The object to be compared with.
+    Returns:
+      True if the objects are equal.
+    """
+    if not isinstance(other, EmbeddingResult):
+      return False
```
Comment (khanhlvg) [inline tensorflow_lite_support/python/task/processor/proto/embedding_pb2.py:156]: Please compare values as well.

**Review by khanhlvg — CHANGES_REQUESTED** [review]

File: tensorflow_lite_support/python/test/task/vision/image_embedder_test.py [hunk 4]
```diff
@@ -119,12 +120,9 @@ def test_embed(self, l2_normalize, quantize, with_bounding_box,
...
     self.assertLen(crop_result.embeddings, 1)
     crop_feature_vector = crop_result.embeddings[0].feature_vector
-    if quantize:
-      self.assertLen(image_feature_vector.value_string, 1024)
-      self.assertLen(crop_feature_vector.value_string, 1024)
-    else:
-      self.assertLen(image_feature_vector.value_float, 1024)
-      self.assertLen(crop_feature_vector.value_float, 1024)
+
+    self.assertLen(image_feature_vector.value, 1024)
```
Comment (khanhlvg) [inline tensorflow_lite_support/python/test/task/vision/image_embedder_test.py:124]: Please also assert the data type in image_feature_vector to check if the correct data type is returned depending on the quantize parameter value.

Comment (kinaryml) [reply tensorflow_lite_support/python/test/task/vision/image_embedder_test.py:124]: @khanhlvg I've updated the tests to incorporate the change.

File: tensorflow_lite_support/python/test/task/text/text_embedder_test.py [hunk 5]
```diff
@@ -105,14 +105,21 @@ def test_embed(self, model_name, l2_normalize, quantize, model_file_type,
     def _check_embedding_size(result):
       self.assertLen(result.embeddings, 1)
       feature_vector = result.embeddings[0].feature_vector
-      if quantize:
-        self.assertLen(feature_vector.value_string, embedding_length)
-      else:
-        self.assertLen(feature_vector.value_float, embedding_length)
+      self.assertLen(feature_vector.value, embedding_length)
```
Comment (khanhlvg) [inline tensorflow_lite_support/python/test/task/text/text_embedder_test.py:108]: There's no need to define helper function like this. You can write each assert in one line just like in the audio_embedder_test.py

Comment (kinaryml) [reply tensorflow_lite_support/python/test/task/text/text_embedder_test.py]: @khanhlvg I've simplified the tests now.

**Review by khanhlvg — APPROVED** [review]

**Pull request merged**